### DISCUSSION TITLE
Managing BeamGroups

### DISCUSSION THREAD

**Discussion by olin256** [Audiveris/audiveris#715] [Q&A]
Is there any way the user can "manage" BeamGroup inters (create, specify members…)? Using the popup menu, all I can do is delete them, after which they simply reappear at some point.

These issues are motivated by the following file:
[004.zip](https://github.com/Audiveris/audiveris/files/14232419/004.zip)
In measure 6, the two beams between the staves merge in a group all the time, because of which the voices are assigned incorrectly.

**Comment by hbitteur** [comment]
Sorry Alexander for my late answer, which is not a complete answer by the way.

Your worked example shows two beams that meet all the geometric criteria to be part of the same group -- except that these beams do not have any stem in common.

As of today, a BeamGroup can be created:
- during the BEAMS step -- at this moment, stems have not yet been detected ...
- as a side effect of manually inserting a beam

I obviously need to work on this:
- At least provide a way to manually modify a BeamGroup
- Make the OMR engine re-check the BeamGroup instances when the stems get available

Stay tuned :-)

**Reply by olin256** [reply]
Thanks for your answer! So the solution for now could be to make the geometric criteria so strict that situations like this one do not meet them.

Perhaps there could be a switch disabling BeamGroups across multiple staves; this is something which never happens in choral music (and probably other genres, too). The same could be said for a single beam.

**Comment by hbitteur** [comment]
@olin256 

I just pushed commit 665b4ee0b40a6430c169986c662d16c1a5a6aee4 on the development branch.
It fixes the problem in the OMR engine (during the REDUCTION step)

TODO: We still need to implement a way to manually manage such false groups if needed...
